src/Native/System.Device.Gpio.Native/libgpiod.c

```diff
@@ -0,0 +1,175 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <gpiod.h>
+
+/**
+ * @brief Close a GPIO chip handle and release all allocated resources.
+ * @param chip The GPIO chip object.
+ */
+extern void CloseChip(struct gpiod_chip *chip) 
```

Comment:
is the gpiod_chip struct define in gpiod.h? If so, why is it that we need it exposed? Is there a way to hide that by only passing in the integers to the shim, and have the ship translate that into the struct? That way, our PInvoke doesn't have to do all the conversions of types, and instead it will only operate with pin numbers.

Reply:
`is the gpiod_chip struct define in gpiod.h?` - No, its just declared there
You mean don't expose the chip and work with pins directly? Most of the methods called here require the chip. There is methods we can use which not requiring the chip pointer, but they still require chip name, internally those methods open the chip to do its job. If we use those methods:
1. It will be slower as chip will be opened and closed back for each access
2. We will still need chip name all the time, we still need to load chips to load name dynamically